pyo3-build-config/src/impl_.rs

```diff
@@ -654,6 +723,32 @@ impl TargetInfo {
             }
         )
     }
+
+    fn is_cross_compiling_from(&self, host: &str) -> bool {
+        let target_triple = self.to_target_triple();
+
+        // Not cross-compiling if arch-vendor-os is all the same
+        // e.g. x86_64-unknown-linux-musl on x86_64-unknown-linux-gnu host
+        //      x86_64-pc-windows-gnu on x86_64-pc-windows-msvc host
+        let host_target_compatible = host.starts_with(&target_triple)
+        // Not cross-compiling to compile for 32-bit Python from windows 64-bit
+        || (target_triple == "i686-pc-windows" && host.starts_with("x86_64-pc-windows"))
+        // Not cross-compiling to compile for x86-64 Python from macOS arm64
+        || (target_triple == "x86_64-apple-darwin" && host == "aarch64-apple-darwin")
+        // Not cross-compiling to compile for arm64 Python from macOS x86_64
+        || (target_triple == "aarch64-apple-darwin" && host == "x86_64-apple-darwin");
+
+        // Cross compiling if not compiler compatible
+        !host_target_compatible
+    }
+
+    fn is_windows(&self) -> bool {
+        self.os == "windows"
+    }
+
+    fn is_windows_mingw(&self) -> bool {
```

Comment:
Why not `is_windows_gnu`?

Reply:
That's the convention the existing code uses:

https://github.com/PyO3/pyo3/blob/9f60b673b8a41dbd6b6ee5e77c34e5d9b750fd7b/pyo3-build-config/src/impl_.rs#L209-L211

https://github.com/PyO3/pyo3/blob/9f60b673b8a41dbd6b6ee5e77c34e5d9b750fd7b/pyo3-build-config/src/impl_.rs#L1370-L1380